.github/workflows/e2e-tests-scheduled.yaml

```diff
@@ -59,3 +79,66 @@ jobs:
         AZURE_LOCATION: "${{ secrets.AZURE_LOCATION }}"
 
         GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+    - name: 'Clean up on failure'
+      run: |
+        timeout 5m ./ci/e2e-tests/test-cleanup.sh ${{ matrix.test_name }}
+        exit 1 # Force the workflow to fail even if this step succeeds
```

Comment:
q: does this mean that Actions report success if one intermediate step failed, but the subsequent is successful? Asking this b/c it is different in ADO, where if any step fails - the whole run marked as failed.

Reply:
Yes, that's what I saw. A task that runs on `${{ failure() }}` and succeeds marks the whole workflow as succeeded.

To be clear, this is specifically because of `fail-fast: false` and because this is a `${{ failure() }}`-gated task.